### PR TITLE
Diverse improvements (see description below)

### PR DESCRIPTION
**Pull request by Valentin-Laurent**
- Moved comments from regression.py to migration_guide.md
- Put parameter confidence_level before conformity_score Used X_train, y_train, X_conf, y_conf everywhere instead of X, y
- Updated classification.py API (docstrings still to be done)
- Corrected docstrings, and added TODOs

### CODE REVIEW

File: mapie_v1/classification.py [hunk 1]
```diff
@@ -52,6 +49,7 @@ def predict(self, X: ArrayLike) -> NDArray:
         scikit-learn classifiers
         Shape (n_samples,)
         """
+        pass
 
     def predict_sets(
```
Comment (jawadhussein462) [inline mapie_v1/classification.py:54]: Missing Parameter `X: ArrayLike`

File: mapie_v1/classification.py [hunk 2]
```diff
@@ -110,11 +111,12 @@ def predict(self,
...
-            self,
-            X: ArrayLike,
-            agg_scores: Optional[str] = "mean",
-            # How to aggregate the scores by the estimators on test data
-            conformity_score_params: Optional[dict] = None) -> NDArray:
+        self,
+        X: ArrayLike,
+        agg_scores: Optional[str] = "mean",
```
Comment (jawadhussein462) [inline mapie_v1/classification.py:116]: agg_scores c'est aggregation_method

**Review by jawadhussein462 — CHANGES_REQUESTED** [review]

File: mapie_v1/classification.py (hunk 1 above)
Comment (Valentin-Laurent) [reply mapie_v1/classification.py:54]: Already here (not indented) line 44, isn't it?
`def predict(self, X: ArrayLike)`

File: mapie_v1/classification.py (hunk 2 above)
Comment (Valentin-Laurent) [reply mapie_v1/classification.py:116]: Sure

**Review by jawadhussein462 — APPROVED** [review]

**Pull request merged**